tattle/engine.py

```diff
@@ -18,45 +18,166 @@
+        sys.stdout.write(question + ' [Y/n]:')
+        choice = raw_input().lower()
+        if choice == '':
+            # If the user just hits enter, then tattle assumes it's a 'y'.
+            return True
+        elif choice in valid:
+            return valid[choice]
+        else:
+            sys.stdout.write("Please respond with 'y' (=yes) or 'n' (=no).\n")
 
-        raise KeyError(GITHUB_ENV_VARS_DONT_EXIST)
+
+def enforce_github_env_variables():
+    """Checks and enforces Github credentials' environment variables.
+
+    """
+    try:
+        github_user = os.environ[GITHUB_USER]
+        github_pass = os.environ[GITHUB_PASS]
+    except KeyError:
+        if not query_yes_no(NO_PASSWORD_PROMPT):
+            github_user = raw_input('github username:')
+            github_pass = getpass.getpass(prompt='github password:')
+            os.environ[GITHUB_USER] = github_user
+            os.environ[GITHUB_PASS] = github_pass
 
 
 def parse_arguments():
+    """Sets up CLI arguments.
 
+    :return: CLI arguments
+    """
     parser = argparse.ArgumentParser(description=ARGUMENT_PARSER_DESCRIPTION)
     parser.add_argument(CONFIG_PATH_COMMAND_NAME, '-c',
-                        metavar='<CONFIG_FILE>',
-
-                        help=CONFIG_PATH_HELP_TEXT,
-                        required=True)
+                        metavar='<PATH_TO_CONFIG>',
+                        help=CONFIG_PATH_HELP_TEXT)
+    parser.add_argument(ORG_NAME_COMMAND_NAME,
+                        '-g',
+                        metavar='<ORGANIZATION>',
+                        help=ORG_NAME_HELP_TEXT)
+    parser.add_argument(NAME_COMMAND_NAME,
+                        '-n',
+                        nargs='*',
+                        metavar='<NAMES>',
+                        help=NAME_HELP_TEXT)
+    parser.add_argument(JIRA_TEAM_COMMAND_NAME,
+                        '-j',
+                        metavar='<JIRA-TEAM>',
+                        help=JIRA_TEAM_HELP_TEXT)
+    parser.add_argument(JIRA_STATUSES_COMMAND_NAME,
+                        '-s',
+                        nargs='*',
+                        metavar='<JIRA-STATUS>',
+                        help=JIRA_STATUSES_HELP_TEXT)
+    parser.add_argument(OUTPUT_PATH_COMMAND_NAME,
+                        '-t',
+                        metavar='<OUTPUT-PATH>',
+                        help=OUTPUT_PATH_HELP_TEXT)
+    parser.add_argument(THREAD_LIMIT_COMMAND_NAME,
+                        '-l',
+                        metavar='<THREAD-LIMIT>',
+                        type=int,
+                        help=THREAD_LIMIT_HELP_TEXT)
 
     return parser.parse_args()
 
 
+def get_filters_from_args(args):
+    """Creates Filter objects from CLI arguments.
+
+    :param args: CLI arguments.
+    :return: List of created Filter objects.
+    """
+    filters_list = list()
+
+    if args.names:
+        names_list = args.names
+        regular_expressions = [name for name in names_list]
```

Comment:
regular_expressions? Maybe names? Maybe just inject args.names to 'regular_expressions': HERE